lib/cretonne/meta/base/semantics.py

```diff
@@ -56,3 +61,42 @@
         alo << iadd(xlo, ylo),
         ahi << iadd(xhi, yhi),
         a << vconcat(alo, ahi)))
+
+iadd_cout.set_semantics(
+    (a, c_out) << iadd_cout(x, y),
+    Rtl(
+        bvx << prim_to_bv(x),
+        bvy << prim_to_bv(y),
+        bva << bvadd(bvx, bvy),
+        bvc_out << bvult(bva, bvx),
+        a << prim_from_bv(bva),
+        c_out << prim_from_bv(bvc_out)
+    ))
+
+bextend.set_semantics(
+    a << bextend(x),
+    (Rtl(
+        bvx << prim_to_bv(x),
+        bvy << bvzeroext(bvx),
```

Comment:
I don't know if the bitwise representation of the Boolean types gets exposed anywhere, but this should probably be a `bvsignext`. The Boolean types are represented as the values 0/-1 in registers.

Only a b1 is represented as 0/1 (which I guess is the same as 0/-1 in a 1-bit register)